dapp/dapp.go

```diff
@@ -98,13 +100,25 @@ func New(l *logger.Logger, app *JsonRepresentation, vmModules []module.Module, c
 		cbMod:        cbm,
 	}
 
+	wait := make(chan error, 1)
+
+	// start the DApp async
 	go func() {
```

Comment:
@borjantrajanoski I am not exactly sure what would happen here. Let's say the select times out and we return. Will the garbage collection pick up this go routine and delete it? I am not entirely sure.

Reply:
@florianlenz from what I can see, since 
```
wait := make(chan error, 1)
```
is a buffered channel, the goroutine

```
go func() {
     _, err := vm.Run(app.Code)
     wait <- err
}()
```
Will complete successfully regardless of the select statement timeout, and the resources that the goroutine occupied will be successfully reclaimed.

This makes the current implementation functional and most likely no modifications are needed.

--- 

However, if the channel were to be unbuffered such as
```
wait := make(chan error)
```
Then the goroutine will leak resources in the case of a timeout, because the Select case 
```
case err := <-wait:
``` 
Will never read from the channel, thus the goroutine will exist forever waiting for someone to read from
```
wait <- err
``` 
Which would likely never happen as the method New() creates a brand new channel with a brand new goroutine

Reply:
My worry is actually about the `vm.Run` if you call with with js that like this. `while(true)` it will never return. So this go routine will be forever in the memory, no? So we would need to have a way to "destroy" the go routine after the timeout. I think it's the exact same problem you described (`thus the goroutine will exist forever waiting for someone to read from`).